lib/features/base_view_model_container.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_mobx/flutter_mobx.dart';
+import 'package:lottie/lottie.dart';
+
+import '../design_system/themes/custom_asset.dart';
+import '../design_system/widgets/screen_init_error.dart';
+import 'base_view_model.dart';
+
+class BaseViewModelContainer extends StatefulWidget {
+  final BaseViewModel viewModel;
+  final Widget child;
+
+  const BaseViewModelContainer({
+    super.key,
+    required this.viewModel,
+    required this.child,
+  });
+
+  @override
+  State<BaseViewModelContainer> createState() => _BaseViewModelContainerState();
+}
+
+class _BaseViewModelContainerState extends State<BaseViewModelContainer> {
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((_) => widget.viewModel.runInit());
+  }
+
+  @override
+  void dispose() {
+    widget.viewModel.dispose();
+    super.dispose();
+  }
+
+  void closeKeyboard(BuildContext context) => FocusScope.of(context).unfocus();
+
+  Widget get loadingScreen => Container(
+        height: MediaQuery.of(context).size.height,
+        decoration: BoxDecoration(
+          color: Colors.white30.withOpacity(0.7),
```

Comment:
We probably wanna move this to a CustomColors or similar class/enum.

Reply:
Yes I created a ticket for the CustomXXX classes so we could do that separately